SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SecretSanta.Business;
+using SecretSanta.Data;
+
+namespace SecretSanta.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class GiftController : ControllerBase
```

Comment:
You could create an "EntityController" that your gift, user, and group controllers inherit from. This EntityController class would inherit from ControllerBase and then have the full implementations of the HTTP requests (get,put,post,delete). Since the implementation of the HTTP requests is essentially the same in all of the controllers, all of your controllers could just use the code in the EntityController class. This behavior was specified as optional for the assignment though.
```suggestion
    public class GiftController : EntityController<Gift>
```